modules/vpc/main.tf

```
@@ -0,0 +1,199 @@
+
+
+#Figure out which availability zones are in the region.
+data "aws_availability_zones" "available_azs" {
+  state = "available"
+  filter {
+    name = "opt-in-status"
+    # Currently, no support for Local Zones, Wavelength, or Outpost
+    values = ["opt-in-not-required"]
+  }
+}
+
```

```suggestion
```
This doesn't seem used.

---

```suggestion
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "~> 4.0"
    }
  }
}
```
tflint says that the aws provider isn't used in the root module when the unused aws_availability_zones is removed. The suggestion is that each module should specify its own provider requirements for each provider it uses. https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.5.0/docs/rules/terraform_unused_required_providers.md

---

`If the provider is used in one or more child modules and you'd prefer to define a single requirement, you can ignore the warning:` 
We will be using aws in more than one child module. Lets keep it in root module. With all of these changes, not seeing any errors/warnings in tflint

---

Without this, there's 3 issues from the linter:
```
$ tflint --recursive
3 issue(s) found:

Warning: terraform "required_version" attribute is required (terraform_required_version)

  on  line 0:
   (source code not available)

Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.5.0/docs/rules/terraform_required_version.md

Warning: provider 'aws' is declared in required_providers but not used by the module (terraform_unused_required_providers)

  on main.tf line 13:
  13:     aws = {
  14:       source  = "hashicorp/aws"
  15:       version = "~> 4.0"
  16:     }

Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.5.0/docs/rules/terraform_unused_required_providers.md

Warning: Missing version constraint for provider "aws" in `required_providers` (terraform_required_providers)

  on modules/vpc/main.tf line 176:
 176: resource "aws_route_table_association" "aap_infrastructure_subnet_association" {

Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.5.0/docs/rules/terraform_required_providers.md
```

Adding the terraform section above cleans up 2 of them and the other can be fixed by removing the unused provider. These modules seem very similar to Ansible's roles or collections in that they should be self contained and have everything in them that they need, so the provider definition seems like it should be in the module.